client/llb/source.go

```diff
@@ -645,6 +650,12 @@ func Chown(uid, gid int) HTTPOption {
 	})
 }
 
+func Accept(accept string) HTTPOption {
```

Comment:
Do we think there will be more headers that will be needed like this in the future? In that case I think it would be better to do `HTTPHeader struct` with predefined key option (we don't want to allow all keys with a map though as we have lots of opinionated logic like `HEAD` requests and `eTags` that we don't want users to mess with). There was some discussion about configurable `UserAgent` in the past (atm we made it use buildkit-specific value by default).

Otoh in another PR that adds `Authorization` header support I wouldn't mix it here as that logic is different and connected to loading secrets.

@jedevc

Reply:
Thanks for the review, @tonistiigi. Initially, I had thought about making it more general to support other headers, but I couldn't think of another request header that would vary the (decoded) response body and wasn't for auth, transfer encoding, or cache control (like you said, I wouldn't want people messing with those).

That said, I'm happy to refactor it to support other headers if you think that's called for.

Reply:
I think option 3 for wire and `llb.Header` for LLB but don't allow any header. It can take a struct instead of a map. Latter is so that we don't have many top-level functions in the future where it is hard to understand that they control a header value for a specific HTTP LLB op.

Reply:
That makes sense. I'll go ahead with the refactor.

Reply:
Refactored according to our discussion. I've updated the PR name and description to reflect the general behavior.